shared/js/background/classes/https-redirects.es6.js

```diff
@@ -5,6 +5,14 @@ const constants = require('../../../data/constants')
 const MAINFRAME_RESET_MS = 3000
 const REQUEST_REDIRECT_LIMIT = 7
 
+/**
+ * This class protects users from accidentally being sent into a redirect loop
+ * if a site we've included into our HTTPS list redirects them back to HTTP.
+ *
+ * Every redirect we perform on a tab gets registered against an instance of this class.
+ * If we hit too many redirects for a request, we block it via canRedirect().
+ */
```

Comment:
I realized this class doesn't have a high level description, so I did a commenting pass here and a few other places.